src/commands/log/mod.ts

```diff
@@ -149,6 +154,68 @@ function fmtSession(idx: string, sess: Session, indentWidth: number) {
     return strings.join("");
 }
 
+function printTimesheet(sessions: Session[]) {
+    // bucket sessions into days
+
+    let buckets: Map<number, number> = new Map();
+    let table: string[][] = [];
+    sessions.forEach((session) => {
+        let start = new Date(session.start);
+        let end = new Date(session.end ? session.end : Date.now());
+        const [startyear, startmonth, startdate] = destructureDate(start);
+        const [endyear, endmonth, enddate] = destructureDate(end);
+
+        const updateBucket = (key: number, val: number) => {
+            let prevVal = buckets.get(key) || 0;
+            buckets.set(key, Math.min(prevVal + val, 24 * 60 * 60 * 1000));
+        };
+
+        if (startdate === enddate) {
+            // session does not spill into another day
+            const key = new Date(startyear, startmonth - 1, startdate).getTime();
+            let hours = end.getTime() - start.getTime();
+            updateBucket(key, hours);
+        } else {
+            // session spans multiple days
+            for (let pointer = new Date(start); pointer <= end; pointer.setDate(pointer.getDate() + 1)) {
+                const key = new Date(pointer.getFullYear(), pointer.getMonth(), pointer.getDate()).getTime();
+                if (pointer.getDay() == start.getDay()) {
+                    // at the beginning
+                    pointer.setHours(23, 59, 59, 999);
+                    updateBucket(key, pointer.getTime() - start.getTime());
+                    pointer.setHours(0, 0, 0, 0);
+                } else if (pointer.getDay() == end.getDay()) {
+                    // at the end
+                    pointer.setHours(0, 0, 0, 0);
+                    updateBucket(key, end.getTime() - pointer.getTime());
+                } else {
+                    // in the middle, hopefully nobody is working 24 hours a day lol
+                    updateBucket(key, 24 * 60 * 60 * 1000);
+                }
+            }
+        }
+    });
+
+    table.push([heading("Month"), heading("Date"), heading("Hours Logged")]);
+    let current_month = -1;
+
+    buckets.forEach((millis, key) => {
+        let bucket_date = new Date(key)
+        let month_string = "";
+        if (current_month != bucket_date.getMonth() - 1) {
+            current_month = bucket_date.getMonth() - 1;
+            month_string = new Intl.DateTimeFormat("en-GB", {month: "long"}).format(bucket_date)
+        }
+        const minutes  = Math.floor(millis / 1000 / 60);
```

Comment:
Why do this instead of using humanReadable(millis, true)?

Reply:
makes sense